### ISSUE TITLE
[Feature Request] Forward Core's metrics to TypeScript-land

### ISSUE THREAD

**Issue by mjameswh** [temporalio/sdk-typescript#1177]
### Describe the solution you'd like

Core is adding support for exporting metrics to lang (see [Core PR](https://github.com/temporalio/sdk-core/pull/544)). This will allow various customizations to metrics handling, including:

- Exporting additional/custom metrics;
- Filtering out some metrics (to reduce cost, since commercial metrics collectors generally charge per metric);
- Export to metrics collectors that are not supported by Core itself.

This ticket is tracking effort at adding support in TS SDK for accepting and handling metrics from Core SDK. The scope of this ticket does not extend to the development of a metric exporter for any specific metrics collector.

**Comment by cretz** [comment]
Until this is also in at least one other core language, I would like to see it marked experimental. I have some ideas about how I think we should expose this lang side (i.e. expose both the ability to impl a custom collector _and_ to emit metrics on the existing one as separate unrelated things), so would like to implement it.

**Comment by mjameswh** [comment]
> Until this is also in at least one other core language, I would like to see it marked experimental.

Yes, definitely

> I have some ideas about how I think we should expose this lang side (i.e. expose both the ability to impl a custom collector and to emit metrics on the existing one as separate unrelated things), so would like to implement it.

No problem, will wait til you're back, so we can discuss details on this.

**Comment by mjameswh** [comment]
> will this feature, once implemented, allow adding custom tags to Temporal metrics (like workflow id, activity id...)

Yes, to some extent. It may not be possible to attach metadata to things that happens inside Core itself.
